google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/v1/MessageDataMatcher.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2017 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.pubsub.v1;
+
+import com.google.protobuf.ByteString;
+import com.google.pubsub.v1.PubsubMessage;
+import org.mockito.ArgumentMatcher;
+
+public class MessageDataMatcher implements ArgumentMatcher<PubsubMessage> {
+
+  private ByteString expectedData;
+
+  public MessageDataMatcher(ByteString expectedData) {
+    this.expectedData = expectedData;
+  }
+
+  @Override
+  public boolean matches(PubsubMessage message2) {
+    return (expectedData == message2.getData());
```

Comment:
Change this to expectedData.equals(message2.getData()) - You don't want to do a reference comparison here (checking that two objects point to the same memory location), you only want to do a content comparison.